vsphere/data_source_vsphere_virtual_machine.go

```diff
@@ -283,6 +288,16 @@ func dataSourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{
 			return fmt.Errorf("error setting guest IP addresses: %s", err)
 		}
 	}
+
+	var vtpmPresent bool
+	for _, dev := range props.Config.Hardware.Device {
+		if _, ok := dev.(*types.VirtualTPM); ok {
+			vtpmPresent = true
+			break
+		}
+	}
+	_ = d.Set("vtpm_present", vtpmPresent)
```

Comment:
```suggestion
	var isVTPMPresentt bool
	for _, dev := range props.Config.Hardware.Device {
		if _, ok := dev.(*types.VirtualTPM); ok {
			isVTPMPresent = true
			break
		}
	}
	_ = d.Set("vtpm_present", isVTPMPresent)
```

A bit more idiomatic as it clearly indicates that it is a boolean variable.